Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -423,27 +423,23 @@ public final class RegistryClient {
                         throw RegistryError.invalidSourceArchive
                     }
 
-                    self.fingerprintStorage.put(package: package,
-                                                version: version,
-                                                fingerprint: .init(origin: .registry(registry.url), value: checksum),
-                                                observabilityScope: observabilityScope,
-                                                callbackQueue: callbackQueue) { storageResult in
-                        switch storageResult {
-                        case .success:
-                            completion(.success(checksum))
-                        case .failure(PackageFingerprintStorageError.conflict(_, let existing)):
-                            switch self.fingerprintCheckingMode {
-                            case .strict:
-                                completion(.failure(RegistryError.checksumChanged(latest: checksum, previous: existing.value)))
-                            case .warn:
-                                observabilityScope.emit(warning: "The checksum \(checksum) from \(registry.url.absoluteString) does not match previously recorded value \(existing.value) from \(String(describing: existing.origin.url?.absoluteString))")
+                    if let fingerprintStorage = self.fingerprintStorage {
+                        fingerprintStorage.put(package: package,
+                                               version: version,
+                                               fingerprint: .init(origin: .registry(registry.url), value: checksum),
+                                               observabilityScope: observabilityScope,
+                                               callbackQueue: callbackQueue) { storageResult in
+                            switch storageResult {
+                            case .success:
                                 completion(.success(checksum))
-                            case .none:
+                            case .failure(let error):
+                                // Don't throw write errors
```

Comment:
out of curiosity, why do we no longer throw in this case?

Reply:
We were doing two types of throw before:
1. checksum has changed
2. actual write failures (e.g., file system, serialization, etc.)

1 is sort of integrity check, and I think it's a bit strange to do it on a write operation when we are already doing it on read.

I suppose we can continue throwing 2 if that's the preference.

Reply:
I'm seeing a failure in Swift CI due to the removed `none` case at https://ci.swift.org/job/swift-PR-Linux-smoke-test/31790/consoleFull#6525590103122a513-f36a-4c87-8ed7-cbc36a1ec144:

```
/home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-main/swiftpm/Tests/WorkspaceTests/RegistryPackageContainerTests.swift:421:39: error: type 'FingerprintCheckingMode' has no member 'none'
01:01:48             fingerprintCheckingMode: .none,
01:01:48                                      ~^~~~
01:01:48 error: fatalError
```